test/sw_mgmt_spec.rb

```diff
@@ -200,4 +218,40 @@
       Registration::SwMgmt.select_addon_products
     end
   end
+
+  describe ".products_from_repo" do
+    before do
+      expect(Yast::Pkg).to receive(:ResolvableProperties)
+        .and_return(YAML.load_file(fixtures_file("products_legacy_installation.yml")))
+    end
+
+    it "Returns product resolvables from the specified repository" do
+      expect(Registration::SwMgmt.products_from_repo(5)).to have(1).item
+    end
+
+    it "Returns empty list if not product is found" do
+      expect(Registration::SwMgmt.products_from_repo(255)).to be_empty
+    end
+  end
+
+  describe ".select_product_addons" do
+    # just the sle-module-legacy product
+    let(:products) { [YAML.load_file(fixtures_file("products_legacy_installation.yml")).first] }
+
+    it "selects remote addons matching the product resolvables" do
+      available_addons = YAML.load_file(fixtures_file("available_addons.yml"))
+
+      # expect the sle-module-legacy product to be selected
+      expect(available_addons[4]).to receive(:selected)
```

Comment:
Um, the tricky part is that the `selected` method adds the Addon to the global list (see https://github.com/yast/yast-registration/blob/master/src/lib/registration/addon.rb#L99) 

The reason for 
```ruby
expect(available_addons[4]).to receive(:selected)
```
is to stub the adding (modifying the global variable).  Without it I'd need to add an `after` block to ensure the list is cleared after the test to avoid possibly side effects for the other tests.